imagePNG.ml

```diff
@@ -1246,6 +1260,21 @@ let write_png fn img =
   in output_idat_from 0;
 
   let iend = { chunk_type = "IEND" ; chunk_data = "" } in
-  write_chunk och iend;
+  write_chunk och iend
+
+end
+
+module PngWrite = PngWriter(Pervasives)
+module PngBufferWrite = PngWriter(Buffer_channel)
+
+let wrap x g f =
+  (try f x with | e -> g x; raise e);
+  g x
+
+let write_png fn img = wrap (open_out_bin fn) close_out (PngWrite.output_png img)
 
-  close_out och
+let string_of_png img =
+  let approx_size = img.width * img.height in
```

Comment:
`Buffer` is limited to `Sys.max_string_size` which is pretty small on some platforms (notably 16 MB on x86).
You need to catch the exception when the internal buffer would exceed this, or document the potential exception in the `mli`.